src/main/java/com/iota/iri/network/pipeline/BroadcastPayload.java

```diff
@@ -0,0 +1,44 @@
+package com.iota.iri.network.pipeline;
+
+import com.iota.iri.controllers.TransactionViewModel;
+import com.iota.iri.network.neighbor.Neighbor;
+
+import java.util.Optional;
+
+/**
+ * Defines a payload which gets submitted to the {@link BroadcastStage}.
+ */
+public class BroadcastPayload {
+
+    private Neighbor originNeighbor;
+    private TransactionViewModel tvm;
+
+    /**
+     * Creates a new {@link BroadcastPayload} with the given neighbor and transaction.
+     * 
+     * @param originNeighbor The neighbor from which the transaction originated from
+     * @param tvm            The transaction
+     */
+    public BroadcastPayload(Neighbor originNeighbor, TransactionViewModel tvm) {
+        this.originNeighbor = originNeighbor;
+        this.tvm = tvm;
+    }
+
+    /**
+     * Gets the origin neighbor.
+     * 
+     * @return empty if there wasn't an origin neighbor
+     */
+    public Optional<Neighbor> getOriginNeighbor() {
+        return originNeighbor == null ? Optional.empty() : Optional.of(originNeighbor);
```

Comment:
I remember @alon-e offered to start using Optional and I agreed and even put it in the style guide.

Now I am kinda sorry I did that, since it causes the creation of more objects and it may not be good for objects that are created frequently since it adds more GC pressure. One can argue that java is heavy with objects anyhows, but I still think we should strive to minimize the amount of objects we create per transaction.

I should add to the styleguide that we should avoid creating `Optional` for classes that we instantiate in large quantities.

Reply:
This `Optional<Neighbor` is there to signal the consumer that the `Neighbor` might be null, because the given payload might not originate from a `Neighbor` but a `broadcastTransactions()` API call. Same as with the `hash of requested tx`.

Reply:
Removed all Optionals from the pipeline.